Router/route_example.js

```diff
@@ -0,0 +1,58 @@
+var loginPageRoute = new Route("login", "login-page"); 
+var userProfileRoute = new Route("user/:name/profile", "home-page");
+var checkLoginRule = RuleRoute.defer((routeData) => UserService.isLoggedin());
+var checkLoginRoute = checkLoginRoute(userProfileRoute);
+var checkLoginMaybe = new RouteMaybe(checkLoginRoute, loginPageRoute);
+
+var components = [
+  // ---->
+  "login-page", 
+  (routeData) => require("login-page"),
+  // ---->
+  "home-page", 
+  (routeData) => require("home-page"),
+  // ---->
+  "user-settings", 
+  (routeData) => require("user-settings-page"),
+  // ---->
+  "user-profile", 
+  (routeData) => require("user-profile-page"),
+  // ---->
+  "user-permissions", 
+  (routeData) => require("user-permissions-page")
+];
+
+var stackRoot = new Route("user", "user-dashboard");
+var userSettings = new Route("user/settings", "user-settings");
+var userPremissions = new Route("user/settings/permissions", "user-permissions");
+var userProfile = new Route("user/profile", "user-profile");
+
+Router.use(stackRoot, userSettings, userPremissions, userProfile);
+// register page components
+Router.register(components);
+
+// register routes
+Router.use(checkLoginMaybe);
+
+//adds new middleware
+Router.use({
+  provide: (routeData, transition, next) => {
+    if (routeData.path.indexOf("user/") === 0 && UserService.getUserRole() === 'admin') {
```

Comment:
Burada admin değilse ne şekilde  davranacağını çok anlamadım. Anladığım:
- Kullanıcı admin ise, route path listesine `/user/admin/home` seçeceğini ekliyor
- Hangi sayfaya yönlenebileceği verilmemiş?

Bu bana daha çok web senaryosu için gerçekçi geliyor; mobil uygulama için değil. Çünkü web'te kullanıcı ya da saldırgan adrese istediğini yazabilir, `/user/admin/home` gibi. Fakat native mobil uygulamada böyle 1 adres yok. Kullanıcı admin değilse, admin sayfalarına gitmesine dair buttonların `visible=false` edilmesi işi görür diye düşünüyorum.
Aynı sebeple mobil uygulamada `404` sayfası da geliştirmezsin.

Reply:
Akis su sekilde
- New route request -> filter with middlewares by injecctin route-data (path, params) -> flnd route by filtered route-data -> get route component -> apply route

Burada milddleware isterse route'u degistirebllir.